### DISCUSSION TITLE
Same endpoint for POST and GET method without the duplicate key error?

### DISCUSSION THREAD

**Discussion by Nikola-Milovic** [OAI/OpenAPI-Specification#2699] [Q&A] [answered]
Hello everyone, I am kinda confused as to why this isn't a thing, or I am I doing this wrong entirely?

I have a user profile path `api/user/profile` with a GET `api/user/profile/{userID}` and POST method on the same path. Why isn't that supported, seems like the usual? I mean both should point to the same profile and only differ whether it's a post or a get method. Is this an issue when using a single YAML file or would the issue persist if I split them up into separate files?

(Using swagger)

**Comment by webron** [comment]
It should be supported, I just find your example a bit unclear, is the path `api/user/profile` or `api/user/profile/{userId}`? Those are two separate paths. Each path can support multiple HTTP methods, so there's no need to duplicate thee path.

**Reply by Nikola-Milovic** [reply]
With `userid`, so then the issue is not in the openapi? As I am getting the duplicate key error for the paths, perhaps its just file type thing? But nevertheless importing it in swagger editor keeps the error.

**Reply by webron** [reply]
Can you share your definition here? Or at least the part(s) that cause the issue?

**Reply by Nikola-Milovic** [reply]
@webron  

```
 /user/profile/{iid}:
    post:
      description: User can create or update their profile.
      security:
        - Bearer: []
      tags:
        - Account
      parameters:
        - in: path
          name: iid
          schema:
            type: string
          required: true
          description: String ID that represents user ID      
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/UserProfileUpdate"
      responses:
        "200":
          description: Success
  /user/profile/{iid}:
    get:
      security:
        - Bearer: []
      tags:
        - Account
      parameters:
        - in: path
          name: iid
          schema:
            type: string
          required: true
          description: String ID that represents user ID
      responses:
        "200":
          description: Success
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/UserProfile"
```

`Parser error duplicated mapping key`

This is in my swagger.yaml

```
openapi: 3.0.1
info:
  title: Test API
  description: Test API
  version: v1
servers:
  - url: /api
  ```

**Reply by webron** [reply]
Right, that's not supported. All you need to do is move both methods under the same path declaration:

```yaml
 /user/profile/{iid}:
    post:
      description: User can create or update their profile.
      security:
        - Bearer: []
      tags:
        - Account
      parameters:
        - in: path
          name: iid
          schema:
            type: string
          required: true
          description: String ID that represents user ID      
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/UserProfileUpdate"
      responses:
        "200":
          description: Success
    get:
      security:
        - Bearer: []
      tags:
        - Account
      parameters:
        - in: path
          name: iid
          schema:
            type: string
          required: true
          description: String ID that represents user ID
      responses:
        "200":
          description: Success
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/UserProfile"
```

**Reply by Nikola-Milovic** [reply]
Oh... Thank you, didnt know that was a thing

**Comment by karenetheridge** [comment]
You've written your definition incorrectly. Both http methods go under the same path entry.

```yaml
openapi: 3.1.0
info:
  title: Test API
  version: 1.2.3
paths:
  /user/profile/{iid}:
    get:
      ...
    post:
      ...
```

**Comment by usern3** [comment]
Is there a way to do this with refs? I haven't been able to figure it out. Something like this (this doesnt work, but is there something similar?

```
paths:
  /task-items/create:
    $ref: 'paths/task-items.create.yml'
  /task-lists:
    allOf:
      - $ref: 'paths/task-lists.get.yml'
      - $ref: 'paths/task-lists.post.yml'
  ```

**Reply by hkosova** [reply, marked as answer]
A path item can contain only one $ref. You'll have to put the GET and POST definitions into the same file so they can be $ref'erenced together:
```yaml
paths:
  /task-items/create:
    $ref: 'paths/task-items.create.yml'
  /task-lists:
    $ref: 'paths/task-lists.yml'   # <------
```

```yaml
# task-lists.yml
get:
  summary: Get task lists
  ...

post:
  summary: Create a task list
  ...
```

**Comment by handrews** [comment]
This seems to have been resolved, closing.

**Comment by shaunyelacorn-godmode** [comment]
paths:
  /task-items/create:
    $ref: 'paths/task-items.create.yml'
  /task-lists:
    $ref: 'paths/task-lists.yml'   # <------
